pkg/ccl/importccl/import_stmt_test.go

```diff
@@ -4540,3 +4542,212 @@ func TestDisallowsInvalidFormatOptions(t *testing.T) {
 		}
 	}
 }
+
+var importSafeFunctions = map[string]struct{}{
+	"current_timestamp": {},
+	"now":               {},
+	"localtimestamp":    {},
+}
+
+// importSafeEvalVisitor is a visitor which verifies if
+// the expression is safe to be used in default expression during import.
+// The returned expression from tree.WalkExpr() is an evaluated
+// expression of appropriate type.
+// NB: This visitor must be invoked only on tree.TypedExpr expressions.
+type importSafeEvalVisitor struct {
+	ctx *tree.EvalContext
+	err error
+}
+
+var _ tree.Visitor = &importSafeEvalVisitor{}
+
+func (v *importSafeEvalVisitor) VisitPre(expr tree.Expr) (bool, tree.Expr) {
+	return v.err == nil, expr
+}
+
+func (v *importSafeEvalVisitor) VisitPost(expr tree.Expr) tree.Expr {
+	// Since we require that the expression we begin with is a tree.TypedExpr,
+	// we no longer have to have these checks here.  We can safely cast
+	// expressions to their typed variants.
+	if fn, ok := expr.(*tree.FuncExpr); ok {
+		resolvedFnName := fn.Func.FunctionReference.(*tree.FunctionDefinition).Name
+		if _, ok := importSafeFunctions[resolvedFnName]; !ok {
+			v.err = errors.Newf("function %q not safe for import", tree.AsString(fn))
+		}
+	}
+
+	if v.err != nil {
+		return expr
+	}
+
+	value, err := expr.(tree.TypedExpr).Eval(v.ctx)
```

Comment:
I don't understand why you're calling eval during the walk -- that doesn't make sense to me.